### PR TITLE
Feature/20 recipe schema

### LINKED ISSUE #20: Design better schema for recipes
Right now, the schema for storing recipes in the db is not very good. We should improve it now so we will have less issues in the future.

### PR DESCRIPTION
**Pull request by KELLOGGS522**
Updated Recipe schema by adding Categories (Fast Food, Indian, Vegetarian, etc...) and Ingredients to Recipes. Ingredients also contain Allergens (Dairy, Nuts, etc...) which will be useful for sorting recipe in the future.

I'm thinking that maybe we should change things to allow to add a recipe as an ingredient to another recipe to make common cooking step easier to add to recipes and maybe allow for a sort of "modular" instructions and calories. 

CreatedByUser row should've been added but since authentification hasn't been done yet it is not possible.

Closes #20

### CODE REVIEW

File: jsuite/jrecipes/models.py [hunk 1]
```diff
@@ -1,16 +1,105 @@
...
+    # TODO ADD ORDERFIELD INT !!! V2 !!!
+
+    class Meta:
+        ordering = ['id']
+
+    def __str__(self):
+        return self.text
+
+class InstructionSubStep(models.Model):
+    # TODO TEST STRING FIELD NAME
```
Comment (JR-prog) [inline jsuite/jrecipes/models.py:37]: was this done?

File: jsuite/jrecipes/models.py [hunk 2]
```diff
@@ -1,16 +1,105 @@
...
+    allergens = models.ManyToManyField("Allergen", blank=True)
+
+    class Meta:
+        ordering = ['-id']
+
+    def __str__(self):
+        return self.name
+
+class UnitType(models.TextChoices):
+    # TODO maybe move UnitType outside so it can be used elsewhere?
```
Comment (JR-prog) [inline jsuite/jrecipes/models.py:59]: this can go now that it was done

File: jsuite/jrecipes/schema/mutations.py [hunk 3]
```diff
@@ -1,9 +1,293 @@
 import graphene
 
-from jrecipes.models import Recipe
-from jrecipes.schema.types import RecipeInput, RecipeType
+from jrecipes.models import Category, Allergen, InstructionStep, InstructionSubStep, Ingredient, IngredientListItem, Recipe
+from jrecipes.schema.types import CategoryInput, CategoryType, AllergenInput, AllergenType, InstructionStepInput, InstructionStepType, InstructionSubStepInput, InstructionSubStepType, IngredientInput, IngredientType, IngredientListItemInput, IngredientListItemType, RecipeInput, RecipeType
 
+#################################################
+#               Category
```
Comment (JR-prog) [inline jsuite/jrecipes/schema/mutations.py:7]: This is okay but didn't we discuss about a certain folder structure that makes this kind of workaround unnecessary? (this comment also goes for the models, types, queries I think)

File: jsuite/jsuite.iml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JR-prog) [inline jsuite/jsuite.iml:1]: Add this to gitignore pls

**Review by JR-prog — CHANGES_REQUESTED** [review]
Very nice PR overall. Just a few things:
- leftover TODO comments
- I don't see the refactoring that we discussed

File: jsuite/jrecipes/schema/mutations.py (hunk 3 above)
Comment (KELLOGGS522) [reply jsuite/jrecipes/schema/mutations.py:7]: See issue #53

**Review by JR-prog — APPROVED** [review]
Looks good.

**Comment by JR-prog** [comment]
Bypassing merge restrictions because e2e tests will be fixed soon, see #47

**Pull request merged**